### PR TITLE
feat:  make the dashboard restart quicker

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes https://emqx.atlassian.net/browse/EMQX-11334
Related PR: https://github.com/emqx/minirest/pull/124

When the dashboard generates a dispatch, it is very slow, taking around 5-10 seconds or more. This results in a delay of 5-10 seconds each time the dashboard is restarted during a cluster join.
With this change, the previous dispatch will be used during the dashboard restart (the dispatch is the same if the emqx version remains unchanged). This will allow the dashboard back to normal without waiting for a new dispatch to be generated.

PS: The dispatch always be regenerated after dashboard restart, The dispatch cache also be updated as expected.

Release version: v/e5.8.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Comment by zmstone** [comment]
Can we compile dispatch to a static module for each version? So it’s going to be static even for the initial boot

**Comment by zhongwencool** [comment]
> Can we compile dispatch to a static module for each version? So it’s going to be static even for the initial boot

Yes, we can use a static way. However, the dynamic way is more flexible for patching minor versions.

**Comment by zmstone** [comment]
> > Can we compile dispatch to a static module for each version? So it’s going to be static even for the initial boot
> 
> Yes, we can use a static way. However, the dynamic way is more flexible for patching minor versions.

How about this: we use persistent_term to store the dynamically compiled dispatches, if exists, we use it, otherwise use the static module.

For upgrade/patching, if youmean patching as running `emqx relup ...` command, then we can add a step to recompile the dispatches after upgraded.
If you mean patching as loading some newer version beams, then the problem still exists, one will need to recompile the dispatches after loading the new beams.

**Comment by zhongwencool** [comment]
> > > Can we compile dispatch to a static module for each version? So it’s going to be static even for the initial boot
> > 
> > 
> > Yes, we can use a static way. However, the dynamic way is more flexible for patching minor versions.
> 
> How about this: we use persistent_term to store the dynamically compiled dispatches, if exists, we use it, otherwise use the static module.
> 
> For upgrade/patching, if youmean patching as running `emqx relup ...` command, then we can add a step to recompile the dispatches after upgraded. If you mean patching as loading some newer version beams, then the problem still exists, one will need to recompile the dispatches after loading the new beams.

Done.
We cache all API rules in the 'priv' directory of emqx_dashboard.
Starting listeners with this cache ensures immediate functionality. Additionally, we regenerate it in emqx_dashboard_listener to ensure that patch changes are applied.

**Review by zmstone — APPROVED** [review]

File: apps/emqx/include/logger.hrl [hunk 1]
```diff
@@ -119,7 +119,13 @@ end).
 -endif.
 
 %% print to 'user' group leader
--define(ULOG(Fmt, Args), io:format(user, Fmt, Args)).
--define(ELOG(Fmt, Args), io:format(standard_error, Fmt, Args)).
+-define(ULOG(Fmt, Args),
```
Comment (zmstone) [inline apps/emqx/include/logger.hrl:122]: This is probably a change which deserves a proper change log, and release in 5.8

File: changes/ce/feat-13242.en.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Significantly increased the startup speed of the EMQX management dashboard.
```
Comment (zmstone) [inline changes/ce/feat-13242.en.md:1]:
```suggestion
Significantly increased the startup speed of EMQX dashboard listener.
```

**Comment by HJianBo** [comment]
I found this PR was merged into the release-57 branch. SO actually it will be released in 5.7.1

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2636)

**Pull request merged**